### PR TITLE
chore: app management typescript

### PR DESCRIPTION
**Pull request by suevalov**
* app-installation and app-definition to typescript
* fix building ESM modules (there was no release yet, so it's not a problem)

### CODE REVIEW

File: lib/entities/app-definition.ts [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+
+interface ArrayFieldType {
+  type: 'Array'
+  items: SingleFieldType | LinkFieldType
+}
+
+type FieldType = SingleFieldType | LinkFieldType | ArrayFieldType
+
+// Locations
+type AppLocation = 'app-config' | 'entry-sidebar' | 'entry-editor' | 'dialog'
```
Comment (giotiskl) [inline lib/entities/app-definition.ts:41]: this definition of `AppLocation` options looks incomplete :thinking: there's also location 'page' now https://www.contentful.com/developers/docs/extensibility/app-framework/locations/

File: typings/generated/entities/app-definition.d.ts [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+interface LinkFieldType {
+    type: 'Link';
+    linkType: LinkType;
+}
+interface ArrayFieldType {
+    type: 'Array';
+    items: SingleFieldType | LinkFieldType;
+}
+declare type FieldType = SingleFieldType | LinkFieldType | ArrayFieldType;
+declare type AppLocation = 'app-config' | 'entry-sidebar' | 'entry-editor' | 'dialog';
```
Comment (giotiskl) [inline typings/generated/entities/app-definition.d.ts:17]: same here 'page' missing

File: lib/entities/app-definition.ts (hunk 1 above)
Comment (suevalov) [reply lib/entities/app-definition.ts:41]: good spot, I copied from `typings`, so it was missing from the beginning :) 

I'll add it.

**Pull request merged**